lib/java/src/main/java/org/apache/thrift/server/AbstractNonblockingServer.java

```diff
@@ -302,6 +302,14 @@ public FrameBuffer(
       }
     }
 
+    /**
+     * Sets the selection key.
+     * @param selectionKey the new key to set.
+     */
+    public void setSelectionKey(SelectionKey selectionKey) {
```

Comment:
Not too sure about this. It feels a bit hacky. Maybe someone who is familiar with the code base can suggest a better solution?

Reply:
I checked this part of the code, and in the current Reactor architecture, all the operations of the selector thread will only happen in the current thread, Inspecting the code shows that only the selector thread processes the selectionKey_ in the frameBuffer and reads and writes to transport, so there are no read or write requests in progress when rebuilding and no visibility issues (you don't need to change SelectionKey to AtomicReference when you strip it of final)

Reply:
My concern is that both `FrameBuffer` and `setSelectionKey` are public, so this could be a footgun in the future.
But based on your comment, I'd say YAGNI for now. I've added a note that it's not tread safe to the javadoc.